core/src/main/java/uk/ac/bbsrc/tgac/miso/core/data/AbstractSequencerReference.java

```diff
@@ -113,14 +170,28 @@ public boolean isDeletable() {
 
   @Override
   public String toString() {
-    StringBuilder sb = new StringBuilder();
-    sb.append(getId());
-    sb.append(" : ");
-    sb.append(getName());
-    sb.append(" : ");
-    sb.append(getFQDN());
-    sb.append(" : ");
-    sb.append(getAvailable());
-    return sb.toString();
+    return "AbstractSequencerReference [id=" + id + ", name=" + name
+        + ", platform=" + (platform == null ? null : platform.getPlatformId()) + ", available=" + available
+        + ", ip=" + (ip == null ? null : ip.toString())
+        + ", serialNumber=" + serialNumber
+        + ", dateCommissioned=" + (dateCommissioned == null ? null : dateCommissioned.toString())
+        + ", dateDecommissioned=" + (dateDecommissioned == null ? null : dateDecommissioned.toString())
+        + ", upgradedSequencerReference=" + (upgradedSequencerReference == null ? null : upgradedSequencerReference.getId()) + "]";
+  }
+  
+  @Override
+  public boolean isActive() {
+    return dateDecommissioned == null;
```

Comment:
implicit reliance on java setting this to null despite it being uninitialized makes me squirm a bit.

From the Java docs:

```
For all reference types, the default value is null.
```

Of course, unless your variable is declared as a local variable.  Then everything breaks.
I guess what I'm saying is I'd sleep much better if you initialized this before attempting a comparison.